examples/games/loading_screen.rs

```diff
@@ -0,0 +1,347 @@
+        }
+    }
+}
+
+// This resource will hold the level related systems ID for later use.
+#[derive(Resource)]
+struct LevelData {
+    unload_level_id: SystemId,
+    level_1_id: SystemId,
+    level_2_id: SystemId,
+}
+
+fn setup(mut commands: Commands) {
+    let level_data = LevelData {
+        unload_level_id: commands.register_one_shot_system(unload_current_level),
+        level_1_id: commands.register_one_shot_system(load_level_1),
+        level_2_id: commands.register_one_shot_system(load_level_2),
+    };
+    commands.insert_resource(level_data);
+
+    // Spawns the UI that will show the user prompts.
+    let text_style = TextStyle {
+        font_size: 50.0,
+        ..default()
+    };
+    commands
+        .spawn(NodeBundle {
+            background_color: BackgroundColor(Color::NONE),
+            style: Style {
+                justify_self: JustifySelf::Center,
+                align_self: AlignSelf::FlexEnd,
+                ..default()
+            },
+            ..default()
+        })
+        .with_children(|parent| {
+            parent.spawn(TextBundle::from_section(
+                "Press 1 or 2 to load a new scene.",
+                text_style,
+            ));
+        });
+}
+
+// Selects the level you want to load.
+fn level_selection(
+    mut commands: Commands,
+    keyboard: Res<ButtonInput<KeyCode>>,
+    level_data: Res<LevelData>,
+    loading_state: Res<LoadingState>,
+) {
+    // Only trigger a load if the current level is fully loaded.
+    if let LoadingState::LevelReady = loading_state.as_ref() {
+        if keyboard.just_pressed(KeyCode::Digit1) {
+            commands.run_system(level_data.unload_level_id);
+            commands.run_system(level_data.level_1_id);
+        } else if keyboard.just_pressed(KeyCode::Digit2) {
+            commands.run_system(level_data.unload_level_id);
+            commands.run_system(level_data.level_2_id);
+        }
+    }
+}
+
+// Marker component for easier deletion of entities.
+#[derive(Component)]
+struct LevelComponents;
+
+// Removes all currently loaded level assets from the game World.
+fn unload_current_level(
+    mut commands: Commands,
+    mut loading_state: ResMut<LoadingState>,
+    entities: Query<Entity, With<LevelComponents>>,
+) {
+    *loading_state = LoadingState::LevelLoading;
+    for entity in entities.iter() {
+        commands.entity(entity).despawn_recursive();
+    }
+}
+
+fn load_level_1(
+    mut commands: Commands,
+    mut loading_data: ResMut<LoadingData>,
+    asset_server: Res<AssetServer>,
+) {
+    // Spawn the camera.
+    commands.spawn((
+        Camera3dBundle {
+            transform: Transform::from_xyz(5.0, 5.0, -5.0)
+                .looking_at(Vec3::new(0.0, 0.0, 0.0), Vec3::Y),
+            ..default()
+        },
+        LevelComponents,
+    ));
+
+    // Save the asset into the `loading_assets` vector.
+    let house_scene = asset_server.load("models/HouseInTheWoods/HouseInTheWoods.glb#Scene0");
```

Comment:
I'd prefer if you used an existing gltf like the fox instead of adding another model to the repo.

Reply:
Will change for the fox, thanks for the feedback.